src/beast/beast/asio/streambuf.h

```diff
@@ -442,6 +442,7 @@ basic_streambuf<Allocator>::prepare (size_type n) ->
                     ++pos;
                     break;
                 }
+                out_end_ = pos->size();
```

Comment:
Here's my understanding: The case this fixes is when we consuming the final block exactly satisfies the prepare request. In that case `out_end_` is unchanged, and may be incorrect due to previous calls to `prepare`.

Ideally, I'd like to check for this case explicitly*. The reason it's coded as an unconditional assignment is that case is hard to check for. We should at least add a comment explaining why we're assigning to out_end_ here.

*This proposal would be useful: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0082r1.pdf